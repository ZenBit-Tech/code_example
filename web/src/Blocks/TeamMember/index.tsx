import React from 'react'; // we need this to make JSX compile
import {
  MemberWrapper,
  MemberWrapperLeft,
  Avatar,
  Name,
  Position,
} from './styles.js';

export type TeamMemberProps = {
  link?: string;
  name: string;
  position: string;
};

export const TeamMember = ({ link, name, position }: TeamMemberProps) => (
  <MemberWrapper>
    <MemberWrapperLeft>
      <Avatar link={link} />
      <Name>{name}</Name>
    </MemberWrapperLeft>
    <Position>{position}</Position>
  </MemberWrapper>
);
