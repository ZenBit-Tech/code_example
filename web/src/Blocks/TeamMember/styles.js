import styled, { css } from 'styled-components';
import AvatarDefault from 'Assets/images/avatar.png';

export const MemberWrapper = styled.div`
  width: 452px;
  height: 100px;
  background: #ffffff;
  border: 1px solid #e9eff4;
  box-sizing: border-box;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0px 20px;
  margin: 10px 0px;
  &:hover {
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.5);
  }
`;
export const MemberWrapperLeft = styled.div`
  display: flex;
`;
export const Avatar = styled.div`
  width: 60px;
  height: 60px;
  background-size: cover;
  background-position: center;
  border-radius: 100%;
  ${(props) =>
    css`
      background-image: url(${props.link ? props.link : AvatarDefault});
    `};
`;
export const Name = styled.div`
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;
  display: flex;
  align-items: center;
  color: #000000;
  margin-left: 12px;
`;
export const Position = styled.div`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  display: flex;
  align-items: center;
  text-align: right;
  color: #666666;
`;
