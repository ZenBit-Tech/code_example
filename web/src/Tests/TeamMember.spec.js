import React from 'react';
import { create } from 'react-test-renderer';
import { TeamMember } from 'Blocks/TeamMember/index';

describe('Team member block', () => {
  test('Team member', () => {
    const memberBlock = create(
      <TeamMember
        link="https://icatcare.org/app/uploads/2018/07/Thinking-of-getting-a-cat.png"
        name="Alexander Morozov"
        position="Manager"
      />,
    );

    const testInstance = memberBlock.root;
    expect(memberBlock.toJSON()).toMatchSnapshot();
    expect(testInstance.findByType(TeamMember).props.position).toBe('Manager');
  });
});
