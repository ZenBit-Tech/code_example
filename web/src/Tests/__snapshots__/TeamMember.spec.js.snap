// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team member block Team member 1`] = `
<div
  className="sc-bdfBwQ chNKQk"
>
  <div
    className="sc-gsTCUz ywWHk"
  >
    <div
      className="sc-dlfnbm hLltOv"
    />
    <div
      className="sc-hKgILt fqfsaa"
    >
      Alexander Morozov
    </div>
  </div>
  <div
    className="sc-eCssSg buEuwN"
  >
    Manager
  </div>
</div>
`;
